### PR TITLE
[CN-539] Add MultiMap CR

### PR DESCRIPTION
**Pull request by cagric0**
New CR created and auto generated files are added.
Phonehome data and sample yaml are added.
[addMultiMapConfig](https://github.com/hazelcast/hazelcast-client-protocol/blob/master/protocol-definitions/DynamicConfig.yaml#L5) codec is added 
Controller, and integration-e2e tests are implemented.

### CODE REVIEW

File: controllers/hazelcast/multimap_controller.go [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+		if err != nil {
+			err = fmt.Errorf("error unmarshaling Last Map Spec: %w", err)
+			return updateMultiMapStatus(ctx, r.Client, mm, mmFailedStatus(err).withMessage(err.Error()))
+		}
+		if !reflect.DeepEqual(&mm.Spec, lastSpec) {
+			return updateMultiMapStatus(ctx, r.Client, mm, mmFailedStatus(fmt.Errorf("MultiMapSpec is not updatable")))
+		}
+	}
+
+	h := &hazelcastv1alpha1.Hazelcast{}
```
Comment (mtyazici) [inline controllers/hazelcast/multimap_controller.go:107]: Can we make this a function? AFAIR, we are using the same code block in the map reconciler.

**Review by hasancelik — CHANGES_REQUESTED** [review]
Can we add multimap removal logic into the Hazelcast CR finalizer [as we did for Map CR before](https://github.com/hazelcast/hazelcast-platform-operator/pull/314)?

File: config/crd/bases/hazelcast.com_multimaps.yaml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    controller-gen.kubebuilder.io/version: v0.4.1
+  creationTimestamp: null
+  name: multimaps.hazelcast.com
+spec:
+  group: hazelcast.com
+  names:
+    kind: MultiMap
+    listKind: MultiMapList
+    plural: multimaps
+    singular: multimap
```
[the remark this answers is not in the document]
Comment (cagric0) [reply config/crd/bases/hazelcast.com_multimaps.yaml:16]: Nice. I was searching where to define shortname :)

[the remark this answers is not in the document]
Comment (mtyazici) [reply config/crd/bases/hazelcast.com_multimaps.yaml:16]: You can do it in the code with, kubebuilder comments. There is an example at `CronHotBackup` CR.

**Review by dzeromski-hazelcast — APPROVED** [review]

**Review by hasancelik — APPROVED** [review]

**Pull request merged**